Babybear

asherbolotski opened the pull request:
gutted smokey code for the preseason bot, removed arm and pneumatic picker stuff and some other unneeded things

asherbolotski committed:
copied over smokey

copied over smokey

asherbolotski committed:
gutted smokey code for baby bear

gutted smokey code for baby bear

mmcadamsms commented on babybear/.gitignore:
```diff
@@ -0,0 +1,162 @@
+# This gitignore has been specially created by the WPILib team.
```
Asher ... you can't checkin in this directory.  Sorry.  Put it in the proper directory please should look like: camp/abolotski/*

Or is this not a mini-project?

josieammer replied:
This is not a mini-project, it's the code for the new pre-season bot.

mmcadamsms replied:
ok

mmcadamsms commented on babybear/.gitignore:
```diff
@@ -0,0 +1,162 @@
+# This gitignore has been specially created by the WPILib team.
```
babybear - Do we have consensus on the name of the robot?  At least from the Seniors (I think they get naming rights on the team).  If not them, how's about the captains?

mmcadamsms replied:
I talked with John.  Seems like name should be 'minibear' ... please rename.

Naher123 replied:
already changed to minibear

mmcadamsms commented on babybear/shuffleboard/BearMaxShuffleBoard.json:
```diff
@@ -0,0 +1,322 @@
+            "size": [
+              2,
+              1
+            ],
```
Are we going to have cone and cube mode in this robot?

Naher123 replied:
removed from JSON

mmcadamsms commented on babybear/shuffleboard/BearMaxShuffleBoard.json:
```diff
@@ -0,0 +1,322 @@
+            "content": {
+              "_type": "Graph",
+              "_source0": "network_table:///SmartDashboard/EveryBotMotorOvercurrentProtection",
+              "_title": "SmartDashboard/EveryBotMotorOvercurrentProtection",
```
Do we need overcurrent protection in this robot?  

Generally ... do we have a block of requirements for this robot?

Naher123 replied:
removed this section of the JSON.

mmcadamsms commented on babybear/shuffleboard/BearMaxShuffleBoard.json:
```diff
@@ -0,0 +1,322 @@
+            "content": {
+              "_type": "Text View",
+              "_source0": "network_table:///SmartDashboard/ExtensionHorizontalArmMeters",
+              "_title": "SmartDashboard/ExtensionHorizontalArmMeters",
```
I can't imagine this is going to be on this robot.

Naher123 replied:
removed this section of the JSON.

mmcadamsms commented on babybear/shuffleboard/BearMaxShuffleBoard.json:
```diff
@@ -0,0 +1,322 @@
+            "content": {
+              "_type": "Boolean Box",
+              "_source0": "network_table:///SmartDashboard/VerticalArmMiddleSensor",
+              "_title": "SmartDashboard/VerticalArmMiddleSensor",
```
Another ... I doubt this is realistic for this robot.

Naher123 replied:
removed this section of the JSON.

mmcadamsms commented on babybear/shuffleboard/BearMaxShuffleBoard.json:
```diff
@@ -0,0 +1,322 @@
+            "content": {
+              "_type": "Boolean Box",
+              "_source0": "network_table:///SmartDashboard/HorizontalArmSensorEncoderEverReset",
+              "_title": "SmartDashboard/HorizontalArmSensorEncoderEverReset",
```
All of these arm things seem out of place for the robot I understand will be built for october.

Naher123 replied:
removed this section of the JSON.

mmcadamsms commented on babybear/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,126 @@
+    // standard stuff constants - motors rotation, etc.
+    public static final double DegreesPerRevolution = 360.0;
+    // NEO maximum RPM
+    public static final double neoMaximumRevolutionsPerMinute = 5676;
```
Are we using neos on this robot?

Naher123 replied:
Yes. Neo is used for the wrist.

mmcadamsms commented on babybear/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,126 @@
+
+    // ***************************************
+    // For auto constants
+    public static final double snoutDepth = Units.inchesToMeters(2.75);
```
This doesn't exist on this robot as best as I'm aware

Naher123 replied:
It is actually used in some auto routines. I've left a TODO in AutonoumousChooser to clean this up. Josie said this is to big to be included in this update.

mmcadamsms commented on babybear/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,126 @@
+
+    // *******************************************************************
+    // Stabilizer constants
+    public static final int StabilizerPneumaticsControlModuleForwardChannel = 0;
```
No pneumatics on this robot AFAIK

Naher123 replied:
Removed from Constants

mmcadamsms commented on babybear/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,126 @@
+
+    // *******************************************************************
+    // picker constants
+    public static final int PickerHorizontalPneumaticsControlModuleForwardChannel = 2;
```
Same as above

Naher123 replied:
Removed from Constants

mmcadamsms commented on babybear/src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,259 @@
+    this.initializeManualInputInterfaces();
+
+    // arm and picker later
+    this.initializeEveryBotPickerSubsystem();
```
So the picker, is it a thing on this robot?

Naher123 replied:
Swapped mention of everybot picker to wrist subsystem

mmcadamsms commented on babybear/src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,259 @@
+              Constants.EveryBotMotorPdpPortId,
+              Constants.EveryBotMotorMaximuCurrentAmps,
+              new SequentialCommandGroup(ebCmd, rc),
+              "EveryBotMotorOvercurrentProtection"));
```
Every bot motor, will we need this on this new robot?

Naher123 replied:
Removed

mmcadamsms commented on babybear/src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,259 @@
+   * A method to init the every bot picker
+   */
+  private void initializeEveryBotPickerSubsystem() {
+    if (InstalledHardware.everyBotPickerInstalled) {
```
More on arm / picker ... see previous comments

Naher123 replied:
Removed

mmcadamsms commented on babybear/src/main/java/frc/robot/commands/EveryBotPickerAutoCommand.java:
```diff
@@ -0,0 +1,105 @@
+import frc.robot.subsystems.EveryBotPickerSubsystem;
+import edu.wpi.first.wpilibj.Timer;
+
+public class EveryBotPickerAutoCommand extends CommandBase {
```
Similar comments as previously posted about picker.  Do we actually know what we need here?

mmcadamsms replied:
seems like to me that minibear should have this file removed

Naher123 replied:
Removed file

mmcadamsms commented on babybear/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,126 @@
+     *
+     * Should be measured from center to center.
+     */
+    public static final double DRIVETRAIN_TRACKWIDTH_METERS = Units.inchesToMeters(17.179);
```
Is this dimension correct for the new robot?

mmcadamsms commented on babybear/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,126 @@
+     *
+     * Should be measured from center to center.
+     */
+    public static final double DRIVETRAIN_WHEELBASE_METERS = Units.inchesToMeters(17.179);
```
Are all of these dimensions correct as per the cad?

mmcadamsms commented on babybear/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,126 @@
+    public static final int FRONT_LEFT_MODULE_DRIVE_MOTOR = 1;
+    public static final int FRONT_LEFT_MODULE_STEER_MOTOR = 2;
+    public static final int FRONT_LEFT_MODULE_STEER_ENCODER = 3;
+    public static final double FRONT_LEFT_MODULE_STEER_OFFSET = -Math.toRadians(147.92 + 344.26 + 307.26);
```
I suspect the steer offset should be set to 0 with a TODO above it

mmcadamsms commented on babybear/src/main/java/frc/robot/commands/EveryBotPickerDefaultCommand.java:
```diff
@@ -0,0 +1,62 @@
+
+import java.util.function.DoubleSupplier;
+
+public class EveryBotPickerDefaultCommand extends CommandBase {
```
Similar comments as previously posted about picker.  Do we actually know what we need here?

Naher123 replied:
File removed

mmcadamsms commented on babybear/src/main/java/frc/robot/common/DebugUtils.java:
```diff
@@ -0,0 +1,83 @@
+
+import java.util.*;
+
+public class DebugUtils {
```
Curious to know if we are using this anymore?

Naher123 replied:
removed

mmcadamsms commented on babybear/src/main/java/frc/robot/control/AutonomousChooser.java:
```diff
@@ -0,0 +1,323 @@
+            autonomousPathChooser.addOption("Node 5 Routine", AutonomousPath.MIDDLE_PATH);
+            autonomousPathChooser.addOption("Node 8 (Right) Routine", AutonomousPath.NODE8_ROUTINE);
+            autonomousPathChooser.addOption("Node 9 (Right) Routine", AutonomousPath.RIGHT_PATH);
+            autonomousPathChooser.addOption("Test Node5 Score Routine", AutonomousPath.TEST_NODE5_SCORE_ROUTINE);
```
Is this score routine expected to work on the new robot?

mmcadamsms commented on babybear/src/main/java/frc/robot/control/AutonomousChooser.java:
```diff
@@ -0,0 +1,323 @@
+    }
+
+    private enum ScoringPosition {
+        SCORE_HIGH,
```
AFAIK - the new robot will only score low ... so middle an especially high should be removed ... unless there was a change in planned robot capabilities.

Naher123 replied:
Robot scores low, mid, and high

mmcadamsms commented on babybear/src/main/java/frc/robot/control/InstalledHardware.java:
```diff
@@ -0,0 +1,58 @@
+    public static final boolean wifiRadioInstalled = true;
+    public static final boolean navx2Installed = false;
+    public static final boolean navx1Installed = true;
+    public static final boolean navxInstalled = navx1Installed || navx2Installed;
```
Conversation has been around only using navx2 for the fall robot.

Naher123 replied:
Changed to say navx2 installed

mmcadamsms commented on babybear/src/main/java/frc/robot/subsystems/CameraSubsystem.java:
```diff
@@ -0,0 +1,53 @@
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class CameraSubsystem extends SubsystemBase {
```
This class seems to not be fully fleshed out.  How to fill in the blanks?

mmcadamsms commented on babybear/src/main/java/frc/robot/subsystems/EveryBotPickerSubsystem.java:
```diff
@@ -0,0 +1,145 @@
+    /* *********************************************************************
+    MEMBERS
+    ************************************************************************/
+    private CANSparkMax everyBotMotor = new CANSparkMax(Constants.EveryBotPickerMotorCanId, MotorType.kBrushless);
```
is this correct for the new fall 2023 robot?

Naher123 replied:
Already changed for wrist motor

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/checksums/checksums.lock:
Asher, I think something is wrong with your .includes.  You likely should fix that and then revert these files.

Under any case:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/checksums/md5-checksums.bin:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/checksums/sha1-checksums.bin:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/dependencies-accessors/dependencies-accessors.lock:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/dependencies-accessors/gc.properties:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/dependencies-accessors/dependencies-accessors.lock:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/build/tmp/jar/MANIFEST.MF:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/build/tmp/compileJava/previous-compilation-data.bin:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/build/libs/EncoderToBagMotor.jar:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/build/jni/release/libwpiutiljni.dylib:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/build/jni/release/libwpiutil.dylib:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/executionHistory/executionHistory.bin:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/fileChanges/last-build.bin:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/executionHistory/executionHistory.lock:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/fileHashes/fileHashes.bin:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/fileHashes/fileHashes.lock:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/fileHashes/resourceHashesCache.bin:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/gc.properties:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/buildOutputCleanup/buildOutputCleanup.lock:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/buildOutputCleanup/cache.properties:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/buildOutputCleanup/outputFiles.bin:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/buildOutputCleanup/outputFiles.bin:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/file-system.probe:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/build/jni/release/libwpinetjni.dylib:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/build/jni/release/libwpinet.dylib:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/build/jni/release/libwpimathjni.dylib:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/build/jni/release/libwpimath.dylib:
Please remove ... this does not belong here.

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/vcs-1/gc.properties:
Ok ... I can't copy paste anymore.  All of these files need to be removed ....

frosty6742 commented:
@asherbolotski either you or I can work on this but your more qualified.

Looking at this...
I see a picker subsystem that sets a neo550 by how much either of the the triggers are pressed. 

According to our cad we have:
 - A wrist: 1 Geared neo550 to move the arm to a up or down position. 
          For this we need two preset positions, maybe A and B on the controller.
          We can have a limit sensor at the stowed position if that's needed. 
 - A Intake: 1 neo550 to run two sets of wheels to pick up a cube
          We need this to be ran by the triggers on a controller (in and out).
          My understanding is that we are only doing cubes so we can remove the cone/cube chooser. 
         
Let me know what you think and what we can do.

frosty6742 committed:
Created Wrist Subsystem

frosty6742 committed:
Create Wrist Default Command

frosty6742 committed:
Wrist Preset Buttons

frosty6742 committed:
Intake Bumpers Added

frosty6742 committed:
Got Wrist Gear Reduction

frosty6742 committed:
Added Wrist Position Tracking

frosty6742 committed:
Added Individual Pickup and Expel speeds depending on position of wrist

frosty6742 committed:
Update Intake Subsystem to use srx talons

frosty6742 committed:
Added Intake Untill Current Spike, will need to tune constant

frosty6742 committed:
Organize, Made Constants

frosty6742 committed:
Update estimated ticks per degree for wrist

frosty6742 committed:
Updated Drive Code

mmcadamsms commented on babybear/shuffleboard/BearMaxShuffleBoard.json:
```diff
@@ -0,0 +1,322 @@
+{
```
this file should be renamed to MiniBearShuffleboard.json

mmcadamsms commented on babybear/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,150 @@
+    public static final double SHOOT_SPEED_2 = 1.0;
+
+
+    public enum WristPosition {
```
Enums should be in their own file.  Please move to Common folder.

mmcadamsms replied:
Use the pattern of ChargedUpGamePiece.java if you aren't familiar with how to do it.

mmcadamsms commented on babybear/src/main/java/frc/robot/commands/AllStopCommand.java:
```diff
@@ -0,0 +1,52 @@
+    public void execute() {
+        if (this.subsystems.getEveryBotPickerSubsystem() != null) {
+            this.subsystems.getEveryBotPickerSubsystem().setPickerRelativeSpeed(0.0);
+        }
```
Owen - I imagine the new wrist needs to be here.

mmcadamsms commented on babybear/src/main/java/frc/robot/commands/WristDefaultCommand.java:
```diff
@@ -0,0 +1,67 @@
+     * @param targetAngle The target angle in degrees for the wrist.
+     */
+
+    public WristDefaultCommand(WristSubsystem wristSubsystem, double targetAngle) {
```
Shouldn't the default command know the default target angle?  Seems like that is a constant that the default command already knows.  Now if you wanted to rename the class to something like 'WristPositionCommand' (instead of Default) then that type of class would likely take a target angle in the ctor.  Better yet would be to take the preset values associate with the wrist position enumeration.

mmcadamsms replied:
so if it were me I'd have a ctor that looks like:
public WristPositionCommand(WristSubsystem wristSubsystem, WristPosition targetPosition)

mmcadamsms replied:
One other comment here is you likely want two commands:
1. a command that drives the wrist to preset positions
2. a command that will manually manipulate the wrist upward/downward based on joystick or trigger type input ... so we can dial in fine granular values for the encoders.

Naher123 replied:
already changed

mmcadamsms commented on babybear/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,150 @@
+    }
+
+
+    public static WristPosition currentWristPosition = WristPosition.PICKUP; // default
```
lets hold this information in the subsystem instead of up here in constants ... constants implies stuff is not writeable here

Naher123 replied:
information was already moved to subsystem

mmcadamsms commented on babybear/src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,150 @@
+    public enum WristPosition {
+           PICKUP,
+           POSITION_1,
+           POSITION_2,
```
Ideally, these should be named PickUp, PositionOne, PositionTwo

mmcadamsms replied:
Even better than PositionOne would be a name something like Transport (as in the transport position). Another might be a name like Stow and this would be the position the wrist should be at during times when the wrist should 'stow' the game piece.  There might be other positions, but the names are meaningful and when someone discusses the robot and uses the 'proper terminology' then everyone involved knows what is happening ... and the code reader is able to see that the same 'proper terminology' is used in the code.

Without things like this in the code, chaos ensues.  Enums were explicitly introduced into languages like Java and C# (and its predecessors like #DEFINE in C/C++) to avoid the problems humans have doing mental translation when 'Number1' might really mean '777'.

frosty6742 replied:
We need to figure out what the different positions will be and what they do, these are a placeholder currently. Added a TODO to figure out good names for them.

mmcadamsms commented on babybear/src/main/java/frc/robot/commands/WristDefaultCommand.java:
```diff
@@ -0,0 +1,67 @@
+    @Override
+    public void execute() {
+        
+        wristSubsystem.setPickerAngle(targetAngle);
```
I suspect that the subsystem might want to understand what WristPosition enums mean.  This is not required however, and instead what you can do is have this class understand what WristPosition enums mean and have this class translate to various setpoints.

If you want this class to be the mediator of what 'WristPosition' means then code would look something like:

    @Override
     public void execute() {

        // Update the state based on target angle (this assumes unique angles for each position)
        double targetAngle = 0.0; // ideally this would be a private static readonly value in this class
        if(theTargetPosition == WristPosition.PickUp) {
             targetAngle = 0.0; // ideally this would be a private static readonly value in this class
        } else if(targetAngle == WristPosition.PositionOne) {
            targetAngle = 30.0; // ideally this would be a private static readonly value in this class
        } else if(targetAngle == Constants.PositionTwo) {
            targetAngle = 60.0; // ideally this would be a private static readonly value in this class
        }
        wristSubsystem.setPickerAngle(targetAngle);  // I do not like the wrist subsystem having methods with 'Picker' in them ... its confusing
    }

As an aside, the if/elseif blocks that follow seem to not do too much.

mmcadamsms commented on babybear/src/main/java/frc/robot/subsystems/WristSubsystem.java:
```diff
@@ -0,0 +1,172 @@
+      wristMotor.set(this.requestedWristMotorSpeed);
+    }
+
+    public void setPickerAngle(double wristAngle) { 
```
Lets change the name to not include 'picker' ... lets use setAngle() or setWristAngle()

mmcadamsms commented on babybear/src/main/java/frc/robot/subsystems/WristSubsystem.java:
```diff
@@ -0,0 +1,172 @@
+        }
+    }
+
+    
```
I think you will need some sort of up/down incremental change type method here too ... moveWristUp, moveWristDown or just moveWrist(double speed)

mmcadamsms commented on babybear/src/main/java/frc/robot/subsystems/WristSubsystem.java:
```diff
@@ -0,0 +1,172 @@
+    private boolean motorInitializedForSmartMotion = false;
+
+    private boolean isWristMotorInverted = false;
+    private double requestedWristMotorSpeed = 0.0;
```
So I think there are 3 motors planned for the 'wrist subsystem' if I'm not mistaken.
1 - neo motor (spark max controller) for the wrist movements - this will have encoder and will have pickup and stow positions (plus maybe some others)
2 - bag motors (STX controller) that are geared the same that will drive the intake/outtake 'cube picker' 

so seems like there is more needed here especially around what needs to happen for the 'cube picker' dual bag motors

mmcadamsms commented on babybear/src/main/java/frc/robot/subsystems/WristSubsystem.java:
```diff
@@ -0,0 +1,172 @@
+     * @param everyBotPickerSpeed the relative speed -1.0 to 1.0 to run the everyBot arm motor at
+     */
+
+    public void setPickerRelativeSpeed(double everyBotPickerSpeed) {
```
I suspect we should have it take an enum something like intake / discharge and then the subsystem knows the proper speed to run the motors at based on empirical testing

mmcadamsms commented on babybear/src/main/java/frc/robot/subsystems/WristSubsystem.java:
```diff
@@ -0,0 +1,172 @@
+
+    public void setPickerRelativeSpeed(double everyBotPickerSpeed) {
+      this.requestedWristMotorSpeed = MotorUtils.truncateValue(everyBotPickerSpeed, -1.0, 1.0);
+      wristMotor.set(this.requestedWristMotorSpeed);
```
nope ... line above will need to be changed to reference the intake outtake motors properly, depending on motor/gearbox mounting side, if they are on the same side they will need to run in opposite directions.

mmcadamsms commented on babybear/src/main/java/frc/robot/subsystems/WristSubsystem.java:
```diff
@@ -0,0 +1,172 @@
+        kFF = 0.00001; 
+        kMaxOutput = 1; 
+        kMinOutput = -1;
+        maxRPM = Constants.neoFiveFiveZeroMaximumRevolutionsPerMinute;
```
ok ... this is for a 550 neo ... not the larger neo that will be used for the wrist ... at least I'm under the impression that the large neo will be used for the wrist ... you tell me

mmcadamsms replied:
you want ... neoMaximumRevolutionsPerMinute if we are using the larger (more powerful neos)

mmcadamsms commented on babybear/src/main/java/frc/robot/subsystems/WristSubsystem.java:
```diff
@@ -0,0 +1,172 @@
+    private void initializeMotorsSmartMotion() {
+      if(this.motorInitializedForSmartMotion == false) { 
+        // PID coefficients
+        kP = 2e-4; 
```
Need to work on the coefficients ... in testing

Naher123 replied:
coefficients were fixed

frosty6742 committed:
Fixes Set 1

frosty6742 committed:
Rename Everybot Picker Subsystem to Intake Subsystem for legibility

frosty6742 committed:
Update AllStopCommand.java

mmcadamsms committed:
Updates for owen

mmcadamsms committed:
merge it

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/build/jni/release/libapriltag.dylib:
This stuff MUST NOT COME IN!!!!

mmcadamsms commented on camp/abolotski/EncoderToBagMotor/.gradle/7.5.1/fileHashes/fileHashes.bin:
Must not come in!!!!

mmcadamsms commented on minibear/src/main/java/frc/robot/subsystems/WristSubsystem.java:
```diff
@@ -0,0 +1,241 @@
+          wristPidController.setReference(targetPositionTicks, ControlType.kPosition);
+      } else {
+          // If within tolerance, stop the wrist motor
+          wristMotor.set(0);
```
Owen - I'm wondering if it will droop when we set this?  Should test if break mode is good enough ... or if the motor needs to constantly be attempting to find the target location.

mmcadamsms commented:
> @asherbolotski either you or I can work on this but your more qualified.
> 
> Looking at this... I see a picker subsystem that sets a neo550 by how much either of the the triggers are pressed.
> 
> According to our cad we have:
> 
> * A wrist: 1 Geared neo550 to move the arm to a up or down position.
>   For this we need two preset positions, maybe A and B on the controller.
>   We can have a limit sensor at the stowed position if that's needed.
> * A Intake: 1 neo550 to run two sets of wheels to pick up a cube
>   We need this to be ran by the triggers on a controller (in and out).
>   My understanding is that we are only doing cubes so we can remove the cone/cube chooser.
> 
> Let me know what you think and what we can do.

mmcadamsms commented:
> > @asherbolotski either you or I can work on this but your more qualified.
> > Looking at this... I see a picker subsystem that sets a neo550 by how much either of the the triggers are pressed.
> > According to our cad we have:
> > 
> > * A wrist: 1 Geared neo550 to move the arm to a up or down position.
> >   For this we need two preset positions, maybe A and B on the controller.
> >   We can have a limit sensor at the stowed position if that's needed.
> > * A Intake: 1 neo550 to run two sets of wheels to pick up a cube
> >   We need this to be ran by the triggers on a controller (in and out).
> >   My understanding is that we are only doing cubes so we can remove the cone/cube chooser.
> > 
> > Let me know what you think and what we can do.

will we really use the Neo550?  Not the larger NEO?

mmcadamsms committed:
Update to approximate current arm position under manual use conditions

mcadamswa committed:
include vendor deps

mcadamswa committed:
Merge branch 'babybear' of https://github.com/Team4682CyBears/Crescendo2024 into babybear

frosty6742 committed:
Fix Intake Getting Stuck on

frosty6742 committed:
Fixed Intake shooting, when told to intake

frosty6742 committed:
Updated vendordeps because it wasnt building. Builds now.

frosty6742 committed:
Added Wrist Encoder Zeroing and Button

frosty6742 committed:
Init Wrist On Robot Startup

frosty6742 committed:
Update MiniBearShuffleBoard.json

frosty6742 committed:
Tune Arm v1

frosty6742 committed:
Swerve Zero

frosty6742 committed:
Update Drive Speed

mcadamswa committed:
add navx2 fix from bear max

add navx2 fix from bear max

mcadamswa committed:
resolve merge conflicts

resolve merge conflicts

mmcadamsms commented on minibear/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java:
```diff
@@ -0,0 +1,954 @@
+  );
+
+  // private byte navxSampleRate = InstalledHardware.navx2Installed? (byte) 50 : (byte) 200;
+  private byte navxSampleRate = (byte) 50;
```
Good ... that article says 50 or 66 is right here.  According to the article, any other values just add drama.

frosty6742 committed:
Wrist Speed Fix

mmcadamsms committed:
Remove a bunch of jibberish

frosty6742 committed:
Added an intake auto command.

The command moves the arm to a shoot position and shoots.

Implemented this in the auto chooser where bear max had the score command.

frosty6742 committed:
Update IntakeAutoCommand.java

mmcadamsms commented on minibear/src/main/java/frc/robot/commands/IntakeAutoCommand.java:
```diff
@@ -0,0 +1,110 @@
+
+        }
+
+        done = true;
```
Owen - you need to protect this and various other things above (like each of the commands) from changing until the execution of each is done.  The method 'execute()' is a non-blocking call, that means that it is supposed to be called every 20 milliseconds and while each of the actions is progressing, the execute() method needs to keep some amount of state around how each of the actions it is doing is progressing.  Only when each of the actions is complete should it set 'done = true'.  Does that make sense?  Do you want help here on how to protect each of the items here?

frosty6742 replied:
Oh, that makes sense. Do I need checks for each or can I just get rid of that done = true; line? Its fine if its just in the isFinished() method right?

mmcadamsms commented on minibear/src/main/java/frc/robot/commands/IntakeAutoCommand.java:
```diff
@@ -0,0 +1,110 @@
+    @Override
+    public void execute() {
+        if (timer.hasElapsed(moveToPositionTimeStamp)) {
+            new WristPositionCommand(wristSubsystem, WristPosition.PositionTwo);
```
its my understanding that two commands needing the same subsystem can execute at the same time ... you can, however just use the subsystem directly

Naher123 replied:
was already changed

frosty6742 committed:
Updated Intake Auto Command to only use the intake.

We are just going to shoot position 3 (backshoot).

mcadamswa committed:
zeroed swerve

mmcadamsms committed:
Update to help auto be a bit more appropriate

mmcadamsms committed:
Merge branch 'babybear' of https://github.com/Team4682CyBears/Crescendo2024 into babybear

frosty6742 committed:
Wrist Speeds Fixed and Scoring Calibrated!

mmcadamsms committed:
Update of angular acceleration based on ask from John in text

mmcadamsms committed:
Update based on request from John - revert is easy if we don't like the behavior

mcadamswa committed:
Update Drive for John

mcadamswa committed:
updated pdh can id

frosty6742 committed:
Ponder Cube Command

frosty6742 committed:
Fix Ponder

josieammer commented:
Closing this PR since the changes were reviewed on a different PR.
